axelor-account/src/main/java/com/axelor/apps/account/service/batch/BatchAutoMoveLettering.java

```diff
@@ -0,0 +1,486 @@
+      companyPartnerDebitMoveLineList.sort(Comparator.nullsLast(moveLineComparator));
+
+      reconcileWithMethod(
+          companyPartnerDebitMoveLineList, companyPartnerCreditMoveLineList, reconcileMethodSelect);
+    }
+  }
+
+  protected void reconcileWithMethod(
+      List<MoveLine> debitMoveLines, List<MoveLine> creditMoveLines, int reconcileMethodSelect) {
+    if (CollectionUtils.isNotEmpty(debitMoveLines) && CollectionUtils.isNotEmpty(creditMoveLines)) {
+
+      BigDecimal debitTotalRemaining =
+          debitMoveLines.stream()
+              .map(MoveLine::getAmountRemaining)
+              .reduce(BigDecimal::add)
+              .orElse(BigDecimal.ZERO);
+      BigDecimal creditTotalRemaining =
+          creditMoveLines.stream()
+              .map(MoveLine::getAmountRemaining)
+              .reduce(BigDecimal::add)
+              .orElse(BigDecimal.ZERO);
+
+      BigDecimal sumDebit =
+          debitMoveLines.stream()
+              .map(MoveLine::getDebit)
+              .reduce(BigDecimal::add)
+              .orElse(BigDecimal.ZERO);
+      BigDecimal sumCredit =
+          creditMoveLines.stream()
+              .map(MoveLine::getCredit)
+              .reduce(BigDecimal::add)
+              .orElse(BigDecimal.ZERO);
+
+      Map<MoveLine, BigDecimal> debitRemaining = new HashMap<>();
+      for (MoveLine debitMoveLine : debitMoveLines) {
+        debitRemaining.put(debitMoveLine, debitMoveLine.getAmountRemaining());
+      }
+      for (MoveLine creditMoveLine : creditMoveLines) {
+        BigDecimal creditRemaining = creditMoveLine.getAmountRemaining();
+        for (MoveLine debitMoveLine : debitMoveLines) {
+          BigDecimal debit = debitMoveLine.getDebit();
+          BigDecimal credit = creditMoveLine.getCredit();
+          BigDecimal nextCreditRemaining = creditRemaining.subtract(debit);
+          BigDecimal nextDebitRemaining = debitRemaining.get(debitMoveLine).subtract(credit);
+          if (nextCreditRemaining.signum() < 0 || nextDebitRemaining.signum() < 0) {
+            continue;
+          }
+
+          boolean reconcileByAmount =
+              reconcileMethodSelect
+                      == AccountingBatchRepository.AUTO_MOVE_LETTERING_RECONCILE_BY_AMOUNT
+                  && debit.compareTo(credit) == 0;
+          boolean reconcileByOrigin =
+              reconcileMethodSelect
+                      == AccountingBatchRepository.AUTO_MOVE_LETTERING_RECONCILE_BY_ORIGIN
+                  && debitMoveLine.getOrigin() != null
+                  && creditMoveLine.getOrigin() != null
+                  && debitMoveLine.getOrigin().equals(creditMoveLine.getOrigin())
+                  && (accountingBatch.getIsPartialReconcile() || debit.compareTo(credit) == 0);
+          boolean reconcileByBalancedMove =
+              reconcileMethodSelect
+                      == AccountingBatchRepository.AUTO_MOVE_LETTERING_RECONCILE_BY_BALANCED_MOVE
+                  && sumDebit.compareTo(sumCredit) == 0;
```

Comment:
I don't know why but it doesn't work. 
I should have almost the same results as the method AUTO_MOVE_LETTERING_RECONCILE_BY_AMOUNT when I test but I get 0 result with the method AUTO_MOVE_LETTERING_RECONCILE_BY_BALANCED_MOVE
Per example, with the method AUTO_MOVE_LETTERING_RECONCILE_BY_AMOUNT, the Moves EP_ACH220197 and EP_ACH220077 are on the same ReconcileGroup, it's ok because they have the same amount. But they should also be on the same ReconcileGroup with the method AUTO_MOVE_LETTERING_RECONCILE_BY_BALANCED_MOVE because at the end, the 2 moves have 0 in their amountRemaining